packages/platforms/accton/arm64/as4224/src/modules/arm64-accton-as4224-fan.c

```diff
@@ -510,23 +542,31 @@ static int as4224_fan_probe(struct platform_device *pdev)
 	int status = -1;
 	int i = 0;
 
+	data->hwmon_dev = hwmon_device_register_with_info(&pdev->dev,
+											DRVNAME, NULL, NULL, NULL);
```

Comment:
FYI passing `NULL` as chip_info (fourth parameter) is considered wrong in newer kernels (5.19 and later), as of commit https://github.com/torvalds/linux/commit/ddaefa209c4ac791c1262e97c9b2d0440c8ef1d5. Doing so may prevent moving to newer LTS kernels or require rewriting the driver.

The old `hwmon_device_register()` still exists and works though as of Linux 6.4, you just need to live with the warning about the deprecated API at boot.

Reply:
> FYI passing `NULL` as chip_info (fourth parameter) is considered wrong in newer kernels (5.19 and later), as of commit [torvalds/linux@ddaefa2](https://github.com/torvalds/linux/commit/ddaefa209c4ac791c1262e97c9b2d0440c8ef1d5). Doing so may prevent moving to newer LTS kernels or require rewriting the driver.
> 
> The old `hwmon_device_register()` still exists and works though as of Linux 6.4, you just need to live with the warning about the deprecated API at boot.

Just replace hwmon_device_register_with_info() with hwmon_device_register_with_groups()